test/support/otlp_endpoint_test.rb

```diff
@@ -0,0 +1,228 @@
+      assert_nil(ENV.fetch('OTEL_EXPORTER_OTLP_TRACES_ENDPOINT', nil))
+      assert_nil(ENV.fetch('OTEL_EXPORTER_OTLP_METRICS_ENDPOINT', nil))
+      assert_nil(ENV.fetch('OTEL_EXPORTER_OTLP_LOGS_ENDPOINT', nil))
+    end
+
+    def assert_headers_nil
+      assert_nil(ENV.fetch('OTEL_EXPORTER_OTLP_METRICS_HEADERS', nil))
+      assert_nil(ENV.fetch('OTEL_EXPORTER_OTLP_TRACES_HEADERS', nil))
+      assert_nil(ENV.fetch('OTEL_EXPORTER_OTLP_LOGS_HEADERS', nil))
+    end
+
+    it 'wrong formatted SW_APM_SERVICE_KEY OTEL PROTO' do
+      ENV['SW_APM_SERVICE_KEY'] = nil
+      ENV['OTEL_EXPORTER_OTLP_ENDPOINT'] = 'https://otel.collector.na-01.cloud.solarwinds.com:443'
+
+      token_setup
+
+      assert_nil(@endpoint.instance_variable_get(:@token))
+      assert_nil(ENV.fetch('OTEL_EXPORTER_OTLP_HEADERS', nil))
+
+      _(ENV.fetch('OTEL_EXPORTER_OTLP_ENDPOINT', nil)).must_equal 'https://otel.collector.na-01.cloud.solarwinds.com:443'
+
+      assert_entity_endpoint_nil
+      assert_headers_nil
+    end
+
+    it 'correct formatted SW_APM_SERVICE_KEY, auth token set for OTEL PROTO' do
+      ENV['SW_APM_SERVICE_KEY'] = '0123456789abcde0123456789abcde0123456789abcde0123456789abcde1234:my_service'
+      ENV['OTEL_EXPORTER_OTLP_ENDPOINT'] = 'https://otel.collector.na-01.cloud.solarwinds.com:443'
+
+      token_setup
+
+      _(@endpoint.instance_variable_get(:@token)).must_equal '0123456789abcde0123456789abcde0123456789abcde0123456789abcde1234'
+
+      _(ENV.fetch('OTEL_EXPORTER_OTLP_HEADERS', nil)).must_equal 'authorization=Bearer 0123456789abcde0123456789abcde0123456789abcde0123456789abcde1234'
+      _(ENV.fetch('OTEL_EXPORTER_OTLP_ENDPOINT', nil)).must_equal 'https://otel.collector.na-01.cloud.solarwinds.com:443'
+
+      assert_entity_endpoint_nil
+      assert_headers_nil
+    end
+
+    it 'wrong formatted SW_APM_SERVICE_KEY APM PROTO' do
+      ENV['SW_APM_SERVICE_KEY'] = nil
+      ENV['SW_APM_COLLECTOR'] = 'apm.collector.na-01.cloud.solarwinds.com:443'
+
+      token_setup
+
+      assert_nil(@endpoint.instance_variable_get(:@token))
+
+      assert_nil(ENV.fetch('OTEL_EXPORTER_OTLP_ENDPOINT', nil))
+      assert_entity_endpoint_nil
+      assert_headers_nil
+    end
+
+    it 'correct formatted SW_APM_SERVICE_KEY, auth token set for OTEL PROTO resolved from APM COLLECTOR config' do
+      ENV['SW_APM_SERVICE_KEY'] = '0123456789abcde0123456789abcde0123456789abcde0123456789abcde1234:my_service'
+      ENV['SW_APM_COLLECTOR'] = 'apm.collector.na-01.cloud.solarwinds.com:443'
+
+      token_setup
+
+      _(@endpoint.instance_variable_get(:@token)).must_equal '0123456789abcde0123456789abcde0123456789abcde0123456789abcde1234'
+
+      assert_nil(ENV.fetch('OTEL_EXPORTER_OTLP_ENDPOINT', nil))
+      assert_entity_endpoint_nil
+      assert_headers_nil
+    end
+
+    it 'wrong formatted SW_APM_SERVICE_KEY NON-SWO OTLP configured' do
+      ENV['SW_APM_SERVICE_KEY'] = nil
+      ENV['OTEL_EXPORTER_OTLP_ENDPOINT'] = 'http://0.0.0.0:4317'
```

Comment:
we should also have tests to check if the end user has explicitly set an OTLP exporter endpoint that does NOT have the port, things still work.